keras/models.py

```diff
@@ -1357,8 +1406,10 @@ def fit_generator(self, generator, samples_per_epoch, nb_epoch,
             callbacks: list of callbacks to be called during training.
             validation_data: dictionary mapping input names and outputs names
                 to appropriate numpy arrays to be used as
-                held-out validation data.
-                All arrays should contain the same number of samples.
+                held-out validation data, or a generator yielding such
+                dictionaries. All arrays should contain the same number
+                of samples. If a generator, a single yield will be used
```

Comment:
How do you feel about adding a parameter for the number of seen examples to use in evaluating the validation loss?   This would mirror the functionality of the samples_per_epoch parameter.